src/v1_0/rw/cfg.cpp

```diff
@@ -11,6 +11,24 @@ namespace abb :: rws :: v1_0 :: rw :: cfg
         std::string content =
             "filepath=" + resource.directory + "/" + resource.filename + "&action-type=replace";
 
-        client.httpPost(uri.str(), content, {Poco::Net::HTTPResponse::HTTP_NO_CONTENT});
+        POCOResult result = client.httpPost(uri.str(), content, {Poco::Net::HTTPResponse::HTTP_NO_CONTENT});
+
+        std::string progress_id;
+
+        auto const loc = std::find_if(
+            result.headerInfo().begin(), result.headerInfo().end(),
+            [] (auto it) { return it.first == "Location"; });
+
+        if(loc != result.headerInfo().end())
+        {
+            std::string const progress = "/progress/";
+            auto const start_postion = loc->second.find(progress);
+
+            if (start_postion != std::string::npos)
+                progress_id = loc->second.substr(start_postion + progress.size());
+        }
+        else
+            BOOST_THROW_EXCEPTION(ProtocolError {"CFGFile progress id not found."}); // TODO: should even throw exception?
```

Comment:
I think yes, you should throw exception, otherwise you cannot return a `progress_id`.